unraid_mcp/tools/docker.py

```
@@ -1,388 +1,326 @@
+          remove - Remove a container (requires container_id, confirm=True)
+          update - Update a container to latest image (requires container_id)
+          update_all - Update all containers with available updates
+          logs - Get container logs (requires container_id, optional tail_lines)
+          networks - List Docker networks
+          network_details - Details of a network (requires network_id)
+          port_conflicts - Check for port conflicts
+          check_updates - Check which containers have updates available
+        """
+        all_actions = set(QUERIES) | set(MUTATIONS) | {"restart"}
+        if action not in all_actions:
+            raise ToolError(f"Invalid action '{action}'. Must be one of: {sorted(all_actions)}")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)